### PR TITLE
DEVOPS-154755 Changed exception throw by Mono.Posix to more meaningful

### PR DESCRIPTION
**Pull request by KlaudiuszBryjaRelativity**

### CODE REVIEW

**Review by KamilKalitaRel — DISMISSED** [review]

File: lib/Resources.resx [hunk 1]
```diff
@@ -418,4 +418,10 @@ MD5 in property: {2}</value>
   <data name="DestinationPathValidationFailed" xml:space="preserve">
     <value>Destination path '{0}' failed the validation.</value>
   </data>
+  <data name="DirectoryNotFoundException" xml:space="preserve">
+    <value>Attempted to access directory "{0}" that is not on the disk.</value>
```
Comment (mmkowal-rel) [inline lib/Resources.resx:422]: I would change this to "Unable to open directory "{0}". Directory does not exist.

File: lib/Resources.resx [hunk 2]
```diff
@@ -418,4 +418,10 @@ MD5 in property: {2}</value>
...
+  </data>
+  <data name="DirectoryForSymlinkNotFoundException" xml:space="preserve">
+    <value>Attempted to access directory "{0}" for symbolic link "{1}" that is not on the disk.</value>
```
Comment (mmkowal-rel) [inline lib/Resources.resx:425]: "unable to resolve symlink {0} to directory {1}. Directory does not exist.

**Review by mmkowal-rel — APPROVED** [review]
I would change the error messages, apart of that code is ok

**Review by KamilKalitaRel — DISMISSED** [review]

File: test/DMLibTest/Cases/EnumeratorTests.cs [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+#if DOTNET5_4
+        [TestMethod]
+        [TestCategory(Tag.Function)]
+#if DNXCORE50
+        [TestStartEnd()]
+#endif
+        public void ShouldThrowOnLinuxWhenSubDirectoryIsDeletedDuringEnumeration()
+        {
+            // Arrange
+            if (!CrossPlatformHelpers.IsLinux)
```
Comment (Pawel-Andruszkiewicz-Relativity) [inline test/DMLibTest/Cases/EnumeratorTests.cs:87]: Booo, we have nicer solution to handle such scenarios in our tests. Relativity > MS :D

Comment (KlaudiuszBryjaRelativity) [reply test/DMLibTest/Cases/EnumeratorTests.cs:87]: Yes. But XUnit/TestMethod do not have such an attribute to handle platform version (as far as I'm concern).

File: test/DMLibTest/Cases/EnumeratorTests.cs [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+                return ex;
+            }
+
+            return null;
+        }
+
+#region Additional test attributes
+
+        [ClassInitialize]
+        public static void MyClassInitialize(TestContext testContext)
```
Comment (Pawel-Andruszkiewicz-Relativity) [inline test/DMLibTest/Cases/EnumeratorTests.cs:147]: Those `My` prefixes doesn't make much sense to me.

**Review by Pawel-Andruszkiewicz-Relativity — APPROVED** [review]

**Review by KamilKalitaRel — APPROVED** [review]

File: test/DMLibTest/Cases/EnumeratorTests.cs (hunk 4 above)
Comment (KlaudiuszBryjaRelativity) [reply test/DMLibTest/Cases/EnumeratorTests.cs:147]: This is DMLib's. I won't change it now.

**Pull request merged**